### ISSUE TITLE
Research for Launching Neuron from DApp

### ISSUE THREAD

**Issue by IronLu233** [Magickbase/neuron-public-issues#153]
## TL;DR
Deep can launch user local Neuron wallet without it already started. If Neuron not run on the user's PC, it will launch it first and receive the wallet connect URL.

## The implement of deep link

There are two necessary steps, register and listen.

### Register deep link

On the Neuron, we should invoke these code when its main process start.

```ts
      if (process.argv.length >= 2) {
        app.setAsDefaultProtocolClient('neuron', process.execPath, [path.resolve(process.argv[1])])
      } else {
        app.setAsDefaultProtocolClient('neuron')
      }
```

The method `setAsDefaultProtocolClient` sets the current executable as the default handler for a protocol (aka URI scheme). 

#### Extra works on MacOS and Linux

On MacOS, all registered URL scheme should be declared beforehand in MacOS `Info.plist`.
Meanwhile, on Linux, the linux desktop file for the app should include new protocol handler.

By using [electron-builder](https://www.electron.build/api/programmatic-usage.html), These build config options should be added.

```yml
protocols:
  name: 'Neuron'
  # Don't forget to set `MimeType: "x-scheme-handler/deeplink"` for `linux.desktop` entry!
  schemes: ['neuron']
# other config
linux:
  desktop:
    MimeType: "x-scheme-handler/deeplink"
```
### response deep link

Previous step allows DApp  to wake neuron and send the Wallet Connect URL, this step is for handling it.

```ts
function responseDeepLink(event: Electron.Event, url: string) {
    console.log(url)

}

app.on('open-url', responseDeepLink)

if (process.os === 'win32') {
  const gotTheLock = app.requestSingleInstanceLock()
  if (!gotTheLock) {
    app.quit()
  } else {
    app.on('second-instance', (event, commandLine, workingDirectory) => {
      // Someone tried to run a second instance, we should focus our window.
      if (mainWindow) {
        if (mainWindow.isMinimized()) mainWindow.restore()
        mainWindow.focus()
      }
      // the commandLine is array of strings in which last element is deep link url
      // the url str ends with /
       responseDeeplink(event, commandLine.pop().slice(0, -1)!)
    })
  }
}
```
## On DApp
On the Dapp, just use a html `<a>` label, or use `window.open` can pass the wallet connect url to Neuron.
```jsx
<a href={`neuron://connect?url=${url}`}>Connect</a>
window.open(`neuron://connect?url=${url}`)
```

## Known bug
On the app,
